### PR TITLE
Added smart contracts and initial tests.

### PR DESCRIPTION
**Pull request by willjgriff**
Added basic smart contract. 

It includes a permission requirement for calling forward or submitting a submission of a script to prevent spamming where desired. 

Added basic tests, although they're not complete. Someone can finish them otherwise I can next week.

### CODE REVIEW

File: test/helpers/helpers.js [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
-  signature = signature.slice(2) // remove 0x
-  const v = web3.utils.toDecimal(`0x${signature.slice(128, 130)}`)
-
-  return {
-    v: v != 27 && v != 28 ? v + 27 : v,
-    r: `0x${signature.slice(0, 64)}`,
-    s: `0x${signature.slice(64, 128)}`,
-  }
-}
+const advanceTime = web3 => Promise.promisify(function(delay, done) {
```
Comment (fabriziovigevani) [inline test/helpers/helpers.js]: There's an `@aragon/test-helpers` function `timeTravel` which is used for this as well. 
Maybe we can use that one instead so we don't add another dependency?

File: contracts/Delay.sol [hunk 2]
```diff
@@ -1,11 +1,102 @@
...
 
-    function initialize() public onlyInit {
+    bytes32 public constant SET_DELAY_ROLE = keccak256("SET_DELAY_ROLE");
+    bytes32 public constant DELAY_EXECUTION_ROLE = keccak256("DELAY_EXECUTION_ROLE");
+
+    string private constant ERROR_CAN_NOT_EXECUTE = "DELAY_CAN_NOT_EXECUTE";
+    string private constant ERROR_CAN_NOT_FORWARD = "DELAY_CAN_NOT_FORWARD";
+
+    struct DelayedScript {
+        uint256 timeSubmitted;
```
Comment (fabriziovigevani) [inline contracts/Delay.sol]: If we are adding the pause functionality maybe we could change this to `executionTime`  and update the logic around that, as we would have to update it's value when `resume()` is executed ?

**Comment by fabriziovigevani** [comment]
Everything looks great! Just left a couple of comments.

I realized it's missing the cancel and pause functionality.

If it's ok for you i can take a stab at it. Can also finish the remaining tests.

File: test/helpers/helpers.js (hunk 1 above)
Comment (willjgriff) [reply test/helpers/helpers.js]: Have you used it/can you point me to any files that do? I don't think I could get it to work.

Comment (willjgriff) [reply test/helpers/helpers.js]: Copied their implementation and fixed for web3js 1.0.0. The @aragon/test-helpers implementation uses `sendAsync` which got deprecated in favour of `send`. Also removed the dependency.

**Comment by willjgriff** [comment]
@fabriziovigevani You're welcome to finish this off. Will check if anyone else has comments and merge it.

File: contracts/Delay.sol (hunk 2 above)
Comment (willjgriff) [reply contracts/Delay.sol]: Forgot the do this myself so please do it when you work on this!

**Pull request merged**